playground/engine.py

```diff
@@ -0,0 +1,818 @@
+        """
+        Renders a single frame, possibly consisting of multiple passes if any visual effects are toggled on.
+        render() is suitable for offline rendering situations, where high quality is preferred.
+
+        By default, 3dgrt requires only a single pass to render.
+        Toggling on antialiasing and depth of field may require additional samples which require additional passes.
+
+        The returned dictionary of rendered buffers always includes the 'rgb' and 'opacity' buffers.
+        """
+        renderbuffers = self.render_pass(camera, is_first_pass=True)
+        while self.has_progressive_effects_to_render():
+            renderbuffers = self.render_pass(camera, is_first_pass=False)
+        return renderbuffers
+
+    def invalidate_materials_on_gpu(self):
+        """ Marks the materials on GPU as out of date.
+        Materials and textures will be uploaded to the GPU during the next rendering pass.
+        """
+        self.is_materials_dirty = True
+
+    @torch.cuda.nvtx.range("load_3dgrt_object")
+    def load_3dgrt_object(self, object_path, config_name='apps/colmap_3dgrt.yaml'):
+        """
+        Loads a 3dgrt object model from object_path.
+        If object is in .ingp, .ply format, the model will be initialized with a config loaded from config_name.
+        """
+        def load_default_config():
+            from hydra.compose import compose
+            from hydra.initialize import initialize
+            with initialize(version_base=None, config_path='../configs'):
+                conf = compose(config_name=config_name)
+            return conf
+
+        if object_path.endswith('.pt'):
+            checkpoint = torch.load(object_path)
+            conf = checkpoint["config"]
+            if conf.render['method'] != '3dgrt':
+                conf = load_default_config()
+            model = MixtureOfGaussians(conf)
+            model.init_from_checkpoint(checkpoint, setup_optimizer=False)
+            object_name = conf.experiment_name
+        elif object_path.endswith('.ingp'):
+            conf = load_default_config()
+            model = MixtureOfGaussians(conf)
+            model.init_from_ingp(object_path, init_model=False)
+            object_name = Path(object_path).stem
+        elif object_path.endswith('.ply'):
+            conf = load_default_config()
+            model = MixtureOfGaussians(conf)
+            model.init_from_ply(object_path, init_model=False)
+            object_name = Path(object_path).stem
+        else:
+            raise ValueError(f"Unknown object type: {object_path}")
+
+        if object_name is None or len(object_name) == 0:
+            object_name = Path(object_path).stem    # Fallback to pick object name from path, if none specified
+
+        model.build_acc(rebuild=True)
+
+        return model, object_name
+
+    @torch.cuda.nvtx.range("rebuild_bvh (mog)")
+    def rebuild_bvh(self, scene_mog):
+        """ Rebuilds all BVHs used by playground:
```

Comment:
Same comments as other MR; please check comments from other MR. 

Here, would add a note when BVH should be rebuilt.